library/core/src/iter/sources/repeat.rs

```diff
@@ -102,7 +102,11 @@ impl<A: Clone> Iterator for Repeat<A> {
     }
 
     #[track_caller]
-    fn count(self) -> usize {
+    fn fold<B, F>(self, _init: B, _f: F) -> B
```

Comment:
I don't think this is correct since even for an infinite iterator, you would still expect `f` to be called multiple times, which may have visible side-effects.